pkg/util/logging_http_transport.go

```diff
@@ -0,0 +1,74 @@
+// Copyright © 2018 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httputil"
+	"os"
+)
+
+type LoggingHttpTransport struct {
+	transport http.RoundTripper
+	stream    io.Writer
+}
+
+func NewLoggingTransport(transport http.RoundTripper) http.RoundTripper {
+	return &LoggingHttpTransport{transport, nil}
+}
+
+func NewLoggingTransportWithStream(transport http.RoundTripper, s io.Writer) http.RoundTripper {
+	return &LoggingHttpTransport{transport, s}
+}
+
+var SENSITIVE_HEADERS = map[string]bool{
+	"Authorization":       true,
+	"WWW-Authenticate":    true,
+	"Cookie":              true,
+	"Proxy-Authorization": true,
+}
+
+func (t *LoggingHttpTransport) RoundTrip(r *http.Request) (*http.Response, error) {
+	stream := t.stream
+	if stream == nil {
+		stream = os.Stderr
+	}
+	reqCopy := *r
+	reqCopy.Header = make(http.Header, len(r.Header))
+	for k, v := range r.Header {
+		sensitive := SENSITIVE_HEADERS[k]
+		if sensitive {
+			reqCopy.Header.Set(k, "ELIDED")
```

Comment:
Side note: As a non-native speaker, I had to look up "ELIDED", (now I know it ;-). Maybe replace with "******" as this is a common idiom for representing sensitive data in an obfuscated way?

(nothing mission critical though)